### PR TITLE
CNV legend in the matrix fixes

### PR DESCRIPTION
**Pull request by creilly8**
## Description
Addresses the remaining issues from PR #2423, thus fully completing issue #2374 . 

Notable changes: 
1. Combined the opts.data and opts.domain color scale arg into one - opts.domain
2. New helper function to remove outliers from the domain

Test: 
1. [Disco example](http://localhost:3000/?genome=hg38&dslabel=GDC&disco=1&sample=TCGA-AP-A059-01A) - should be no change in rendering
2. [Matrix issue with outliers](http://localhost:3000/?noheader=1&mass=%7B%22dslabel%22:%22ASH%22,%22genome%22:%22hg38%22,%22plots%22:%5B%7B%22chartType%22:%22matrix%22,%22termgroups%22:%5B%7B%22lst%22:%5B%7B%22id%22:%22Sex%22%7D,%7B%22term%22:%7B%22gene%22:%22TAL1%22,%22type%22:%22geneVariant%22%7D%7D,%7B%22term%22:%7B%22gene%22:%22KRAS%22,%22type%22:%22geneVariant%22%7D%7D,%7B%22term%22:%7B%22gene%22:%22MYB%22,%22type%22:%22geneVariant%22%7D%7D,%7B%22term%22:%7B%22gene%22:%22MYC%22,%22type%22:%22geneVariant%22%7D%7D,%7B%22term%22:%7B%22gene%22:%22CDKN2A%22,%22type%22:%22geneVariant%22%7D%7D,%7B%22term%22:%7B%22gene%22:%22NOTCH1%22,%22type%22:%22geneVariant%22%7D%7D%5D%7D%5D%7D%5D%7D) - Outliers should not render in the scale, regardless if the mutations are hidden or gain or loss is hidden. Also note the issue with the overlapping ticks is resolved now that the outliers are removed.
3. Rendering and functionality should not change for any of the other color scale implementations: [hic](http://localhost:3000/?appcard=Hi-C&example=Whole%20genome), [single cell](http://localhost:3000/?noheader=1&mass={%22genome%22:%22hg38%22,%22dslabel%22:%22GDC%22,%22nav%22:{%22header_mode%22:%22hidden%22},%22plots%22:[{%22chartType%22:%22singleCellPlot%22}]}), [mds3 lollipop](http://localhost:3000/?genome=hg38&gene=p16&mds3=ASH), [LD results in the regression analysis](http://localhost:3000/?noheader=1&mass=%7B%22genome%22:%22hg38%22,%22dslabel%22:%22SJLife%22,%22nav%22:%7B%22activeTab%22:1%7D,%22plots%22:%5B%7B%22chartType%22:%22regression%22,%22regressionType%22:%22linear%22,%22outcome%22:%7B%22id%22:%22LV_Ejection_Fraction_3D%22%7D,%22independent%22:%5B%7B%22id%22:%22sex%22,%22refGrp%22:%222%22%7D,%7B%22term%22:%7B%22type%22:%22snplocus%22%7D,%22q%22:%7B%22chr%22:%22chr17%22,%22start%22:7664218,%22stop%22:7671018,%22AFcutoff%22:5,%22alleleType%22:0,%22geneticModel%22:0,%22restrictAncestry%22:%7B%22name%22:%22European%20ancestry%22,%22tvs%22:%7B%22term%22:%7B%22id%22:%22genetic_race%22,%22type%22:%22categorical%22,%22name%22:%22Genetically%20defined%20race%22%7D,%22values%22:%5B%7B%22key%22:%22European%20Ancestry%22,%22label%22:%22European%20Ancestry%22%7D%5D%7D%7D,%22variant_filter%22:%7B%22type%22:%22tvslst%22,%22join%22:%22and%22,%22in%22:true,%22lst%22:%5B%7B%22type%22:%22tvs%22,%22tvs%22:%7B%22isnot%22:true,%22values%22:%5B%7B%22label%22:%22Bad%22,%22key%22:%22Bad%22%7D%5D,%22term%22:%7B%22id%22:%22QC%22,%22name%22:%22Classification%22,%22parent_id%22:null,%22isleaf%22:true,%22type%22:%22categorical%22,%22values%22:%7B%22Good%22:%7B%22label%22:%22Good%22,%22key%22:%22Good%22%7D,%22Bad%22:%7B%22label%22:%22Bad%22,%22key%22:%22Bad%22%7D%7D%7D%7D%7D,%7B%22type%22:%22tvs%22,%22tvs%22:%7B%22ranges%22:%5B%7B%22start%22:0.95,%22startinclusive%22:true,%22stopunbounded%22:true%7D%5D,%22term%22:%7B%22id%22:%22SJcontrol_CR%22,%22name%22:%22SJLIFE%20control%20call%20rate%22,%22parent_id%22:null,%22isleaf%22:true,%22type%22:%22float%22%7D%7D%7D,%7B%22type%22:%22tvs%22,%22tvs%22:%7B%22ranges%22:%5B%7B%22start%22:0.95,%22startinclusive%22:true,%22stopunbounded%22:true%7D%5D,%22term%22:%7B%22id%22:%22CR%22,%22name%22:%22Call%20rate,%20SJLIFE+CCSS%22,%22parent_id%22:null,%22isleaf%22:true,%22type%22:%22float%22%7D%7D%7D,%7B%22type%22:%22tvs%22,%22tvs%22:%7B%22ranges%22:%5B%7B%22start%22:0.95,%22startinclusive%22:true,%22stopunbounded%22:true%7D%5D,%22term%22:%7B%22id%22:%22CR_sjlife%22,%22name%22:%22SJLIFE%20call%20rate%22,%22parent_id%22:null,%22isleaf%22:true,%22type%22:%22float%22%7D%7D%7D,%7B%22type%22:%22tvs%22,%22tvs%22:%7B%22ranges%22:%5B%7B%22start%22:0.95,%22startinclusive%22:true,%22stopunbounded%22:true%7D%5D,%22term%22:%7B%22id%22:%22CR_ccss%22,%22name%22:%22CCSS%20call%20rate%22,%22parent_id%22:null,%22isleaf%22:true,%22type%22:%22float%22%7D%7D%7D,%7B%22type%22:%22tvs%22,%22tvs%22:%7B%22isnot%22:true,%22values%22:%5B%7B%22label%22:%22yes%22,%22key%22:%221%22%7D%5D,%22term%22:%7B%22id%22:%22Polymer_region%22,%22name%22:%22Polymer%20region%22,%22parent_id%22:null,%22isleaf%22:true,%22type%22:%22categorical%22%7D%7D%7D%5D%7D%7D%7D%5D%7D%5D%7D), and [scatter plot](http://localhost:3000/?noheader=1&mass={%22dslabel%22:%22PNET%22,%22genome%22:%22hg19%22,%22plots%22:[{%22chartType%22:%22sampleScatter%22,%22name%22:%22Methylome%20TSNE%22,%22colorTW%22:{%22id%22:%22Age%22,%22q%22:{%22mode%22:%22continuous%22}}}]}).

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: client/dom/ColorScale/ColorScale.ts [hunk 1]
```diff
@@ -31,22 +31,23 @@ export class ColorScale {
...
 		this.ticks = opts.ticks || 5
 		this.tickSize = opts.tickSize || 1
 		this.topTicks = opts.topTicks || false
-		this.domain = opts.domain
+		this.usePercentiles = opts.usePercentiles || false
 
 		this.validateOpts(opts)
-
-		this.tickValues = niceNumLabels(opts.data)
+		if (opts.usePercentiles) opts.domain = removeExtremeOutliers(opts.domain)
```
Comment (xzhou82) [inline client/dom/ColorScale/ColorScale.ts:49]: i'm thinking calculating cutoff value by a percentile should be carried out outside of ColorScale. 

also in cnv legend, caller supplies opts.domain[] array with the axis start/stop, but not the actual long array of cnv values for all cnv events, thus should not do percentile calculation here

**Review by xzhou82 — DISMISSED** [review]
thanks makes sense for colorscale to accept precomputed axis domain values, instead of raw list of hundreds of values from which to figure out axis

may add unit tests to helpers.ts later

@congyu-lu  should review matrix code changes

**Comment by creilly8** [comment]
> image The ticks on color scale should not be from -1 to 1, while a lot of samples have value smaller than -1 (for example CDKN2A)

In master, the range for this scenario is [-8.4, -0.99, -0.98, -0.97 ... 0, ... 0.97, 0.98, 0.99, 8.4]. The changes in this PR remove the -/+ 8.4 values because they are outliers.

**Review by xzhou82 — APPROVED** [review]
works well thanks

**Pull request merged**